src/identity_component.cairo

```diff
@@ -378,20 +383,8 @@ pub mod IdentityComponent {
         fn key_has_purpose(
             self: @ComponentState<TContractState>, key: felt252, purpose: felt252,
         ) -> bool {
-            let key_storage_path = self.Identity_keys.entry(key);
-            if key_storage_path.key.read().is_zero() {
-                return false;
-            }
-            let purposes_storage_path = key_storage_path.purposes.deref();
-            let mut has_purpose = false;
-            for i in 0..purposes_storage_path.len() {
-                let _purpose = purposes_storage_path[i].read();
-                if _purpose == 1 || purpose == _purpose {
-                    has_purpose = true;
-                    break;
-                }
-            };
-            has_purpose
+            let purposes = self.Identity_keys.entry(key).read().purposes;
+            BitmapTrait::get(purposes, 1) || BitmapTrait::get(purposes, purpose.try_into().unwrap())
```

Comment:
indexes 0 and 1 haves defined purposes, maybe we could define them in a constant/enum to improve clarity.